src/lambda/commands/uploadLambda.ts

```diff
@@ -412,3 +441,156 @@ async function uploadZipBuffer(
         localize('AWS.lambda.upload.done', 'Successfully uploaded Lambda function {0}', lambda.name)
     )
 }
+
+export async function findApplicationJsonFile(startPath: vscode.Uri): Promise<vscode.Uri | undefined> {
+    const isTemplateFile = /template.(yaml|yml|json)$/.test(startPath.fsPath)
+    const parentDir = isTemplateFile ? vscode.Uri.joinPath(startPath, '..') : startPath
+
+    if (isCloud9()) {
+        const found = await findFile(parentDir.fsPath, '.application.json')
+        if (!found) {
+            getLogger().debug(
+                'lambda: .application.json file not found in parent directory or sub directories of %s',
+                parentDir.fsPath
+            )
+        }
+        return found ? vscode.Uri.file(found) : undefined
+    } else {
+        const found = await vscode.workspace.findFiles(
+            new vscode.RelativePattern(parentDir.path, '**/*.application.json'),
+            undefined,
+            1
+        )
+        return found[0]
+    }
+}
+
+export function getFunctionNames(file: vscode.Uri, region: string): string[] | undefined {
```

Comment:
almost all functions in this PR do not have documentation. It is worth including a one-line docstring for non-obvious things. Here for example, the non-obvious thing is that this function expects a `.application.json` file.